pkg/runner/expression.go

```diff
@@ -262,7 +266,7 @@ func EvalBool(evaluator ExpressionEvaluator, expr string) (bool, error) {
 			result = t != 0
 		}
 	default:
-		return false, fmt.Errorf("Unable to map return type to boolean for '%s'", expr)
+		return false, fmt.Errorf("Unable to map type '%t' to boolean for '%s'", evaluated, expr)
```

Comment:
I thought the default case is impossible to reach for a conforming evaluator.
Any Object != null => true, except empty string
- `if: github.event.pull_request` ( true if PR )
- `if: fromJSON('{}')` ( always true )

<Details>
<Summary>This weird workflow only fails in act, the last step is expected to run</Summary>

```yaml
on:
- push
- pull_request
jobs:
  m:
    runs-on:
    - ubuntu-latest
    steps:
    - run: |
                echo PR!
      shell: bash
      if: |
                github.event.pull_request
    - run: |
                echo Push!
      shell: bash
      if: |
                !github.event.pull_request
    - run: |
                echo empty obj!
      shell: bash
      if: |
                fromJSON('{}')
```

</Details>

Reply:
Good catch, I'll update the function to handle maps.